### PR TITLE
Migrating Header

### PR DESCRIPTION
**Pull request by Bhammy**
Migrating Header to Prism!

Edit: Also renames `Header` -> `PageHeader`, in hopes of being more semantic about where this component gets used and because we have a number of specific header components across the site.

Screen Shot 2020-08-17 at 4 39 26 PM
Screen Shot 2020-08-17 at 4 39 33 PM
Screen Shot 2020-08-17 at 4 40 06 PM

### CODE REVIEW

File: src/components/Header/Header.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+Very large text that can be used to present the title and short additional intro text for a non-article page.
+
+It takes optional children that should be used for relevant non-text components, such as navigation elements.
+
+## Usage 
+
+```js
+import { Header } from '@quartz/interface/src/components';
+```
+
```
Comment (chriszarate) [inline src/components/Header/Header.stories.mdx:17]: fyi i've been removing this, very redundant. i think i'll add it to the welcome story
```suggestion
```

File: src/components/PageHeader/PageHeader.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+# PageHeader
+
+Very large text that can be used to present the title and short additional intro text for a non-article page.
+
+It takes optional children that should be used for relevant non-text components, such as navigation elements.
+
+## Props
+
+<Props of={PageHeader} />
```
Comment (chriszarate) [inline src/components/PageHeader/PageHeader.stories.mdx:14]: Use `ArgsTable` for a fun interactive props table. (see Example.stories.mdx). This is new in the PR I just merged this morning, sorry.

Comment (Bhammy) [reply src/components/PageHeader/PageHeader.stories.mdx]: Oooooh!

File: src/components/PageHeader/PageHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './PageHeader.scss';
+
+const PageHeader = ( { border, children, intro, title } ) => (
+	<div className={`${styles.container} ${border ? styles.border : ''}`}>
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.jsx:6]: WDYT about using `<header>`? 🤔

File: src/components/PageHeader/PageHeader.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+		<div className={styles.contents}>
+			<h1 className={styles.title}>{title}</h1>
+			<p className={styles.intro}>{intro}</p>
+		</div>
+		{children}
+	</div>
+);
+
+PageHeader.propTypes = {
+	border: PropTypes.bool,
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.jsx:16]:
```suggestion
	border: PropTypes.bool.isRequired,
```

File: src/components/PageHeader/PageHeader.scss [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	text-align: center;
+
+	&.border {
+		border-bottom: $border-solid-decorative;
+	}
+}
+
+.contents {
+	margin-top: 20px;
+	margin-bottom: 25px;
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.scss:16]: Use 4px grid

```suggestion
	margin-bottom: 24px;
```

File: src/components/PageHeader/PageHeader.scss [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+	@include for-tablet-portrait-up {
+		margin-top: 30px;
+		margin-bottom: 50px;
+	}
+
+	@include for-tablet-landscape-up {
+		margin-top: 50px;
+		margin-bottom: 60px;
+	}
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.scss:26]: I suspect these media queries and values are pretty arbitrary, could you check with Daniel whether we can simplify?

File: src/components/PageHeader/PageHeader.scss [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	@include font-maison-800-10;
+
+	margin: 0;
+}
+
+.intro {
+	@include font-maison-500-5;
+
+	max-width: 500px;
+	margin: 7px auto 0 auto;
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.scss:39]:
```suggestion
	margin: 8px auto 0;
```

File: src/components/PageHeader/PageHeader.scss [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+	color: $color-typography-faint;
+
+	@include for-tablet-portrait-up {
+		max-width: 600px;
+	}
+
+	@include for-tablet-landscape-up {
+		max-width: 760px;
+		margin-top: 10px;
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.scss:48]: Seems unnecessary imo

```suggestion
```

File: src/components/PageHeader/PageHeader.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,78 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
+import PageHeader from './PageHeader.jsx';
+
+<Meta title="PageHeader" component={PageHeader} />
+
+# PageHeader
+
+Very large text that can be used to present the title and short additional intro text for a non-article page.
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.stories.mdx:8]: Nitpick but imo 'Very large text' isn't particularly useful info, the 2nd half of this sentence seems more important

File: src/components/PageHeader/PageHeader.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+It takes optional children that should be used for relevant non-text components, such as navigation elements.
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.stories.mdx:10]: We should probably just specify that this should always be a navigational element, e.g. tabs or a search bar. 'Non-text' components could be misleading

File: src/components/PageHeader/PageHeader.jsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+	<div className={`${styles.container} ${border ? styles.border : ''}`}>
+		<div className={styles.contents}>
+			<h1 className={styles.title}>{title}</h1>
+			<p className={styles.intro}>{intro}</p>
+		</div>
+		{children}
+	</div>
+);
+
+PageHeader.propTypes = {
```
Comment (jshakes) [inline src/components/PageHeader/PageHeader.jsx:15]: describe each prop with a docblock (see example component)

File: src/components/PageHeader/PageHeader.stories.mdx (hunk 10 above)
Comment (Bhammy) [reply src/components/PageHeader/PageHeader.stories.mdx:10]: Gotcha, I was looking around and it seemed like it should only be nav stuff but I wasn't sure if this was a hard rule

File: src/components/PageHeader/PageHeader.scss (hunk 5 above)
Comment (Bhammy) [reply src/components/PageHeader/PageHeader.scss:16]: Nice - did we document this somewhere?

File: src/components/PageHeader/PageHeader.jsx (hunk 11 above)
Comment (Bhammy) [reply src/components/PageHeader/PageHeader.jsx:15]: Isn't this what `ArgsTable` is for? I'm not sure what I should be looking for in `Example`

File: src/components/PageHeader/PageHeader.jsx (hunk 3 above)
Comment (Bhammy) [reply src/components/PageHeader/PageHeader.jsx:6]: This sent me down a small rabbit hole. I like it generally, but MDN seems to suggest that `<nav>` should not be a child of `<header>`, which is the most obvious use case of the `children` prop for this component. I don't think that's an unforgivable sin, but would be interested to know what you think.

Comment (jshakes) [reply src/components/PageHeader/PageHeader.jsx:6]: Interesting, I didn't know that. Let's leave it for now, we can always revisit

File: src/components/PageHeader/PageHeader.jsx (hunk 11 above)
Comment (jshakes) [reply src/components/PageHeader/PageHeader.jsx:15]: `ArgsTable` and `PropsTable` pull the `description` of each prop from a comment block above the proptype definition. See https://github.com/Quartz/interface/blob/main/src/components/Example/Example.jsx#L12

Comment (Bhammy) [reply src/components/PageHeader/PageHeader.jsx:15]: Derp! thank you

File: src/components/PageHeader/PageHeader.scss (hunk 6 above)
Comment (danielllee) [reply src/components/PageHeader/PageHeader.scss:26]: margin
```suggestion
	@include for-tablet-portrait-up {
		margin-top: 48px;
		margin-bottom: 60px;
	}

	@include for-tablet-landscape-up {
		margin-top: 48px;
		margin-bottom: 60px;
	}
```

Comment (Bhammy) [reply src/components/PageHeader/PageHeader.scss]: Thank you!

**Pull request merged**